datadog-opentelemetry/src/telemetry_metrics_exporter.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright 2025-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+use opentelemetry_sdk::error::OTelSdkResult;
+use opentelemetry_sdk::metrics::data::ResourceMetrics;
+use opentelemetry_sdk::metrics::exporter::PushMetricExporter;
+use opentelemetry_sdk::metrics::Temporality;
+use std::fmt::Debug;
+use std::time::Duration;
+
+use crate::core::telemetry;
+use crate::metrics_exporter::OtlpProtocol;
+
+pub struct TelemetryTrackingExporter<E> {
+    inner: E,
+    protocol: OtlpProtocol,
+}
+
+impl<E> TelemetryTrackingExporter<E> {
+    pub fn new(exporter: E, protocol: OtlpProtocol) -> Self {
+        Self {
+            inner: exporter,
+            protocol,
+        }
+    }
+}
+
+impl<E> Debug for TelemetryTrackingExporter<E>
+where
+    E: Debug,
+{
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("TelemetryTrackingExporter")
+            .field("inner", &self.inner)
+            .finish()
+    }
+}
+
+impl<E> PushMetricExporter for TelemetryTrackingExporter<E>
+where
+    E: PushMetricExporter,
+{
+    async fn export(&self, metrics: &ResourceMetrics) -> OTelSdkResult {
+        use telemetry::TelemetryMetric::*;
+
+        let (attempts, successes, failures) = match self.protocol {
+            OtlpProtocol::Grpc => (
+                OtelMetricsExportAttemptsGrpc,
+                OtelMetricsExportSuccessesGrpc,
+                OtelMetricsExportFailuresGrpc,
+            ),
+            OtlpProtocol::HttpProtobuf | OtlpProtocol::HttpJson => (
+                OtelMetricsExportAttemptsHttp,
+                OtelMetricsExportSuccessesHttp,
+                OtelMetricsExportFailuresHttp,
+            ),
+        };
+
+        telemetry::add_point(1.0, attempts);
```

Comment:
Is it fine to add one telemetry point per point exported? Or maybe exports are not that frequent?

Reply:
We are adding one telemetry point per export. By default we will export metrics every 10 seconds, so this should be fine for most use cases.